plugins/stringlib/stringlib.cpp

```diff
@@ -1433,7 +1455,7 @@ static const char * simple_strptime(size32_t lenStr, const char * str, const cha
                     offset++;
                 break;
             case 'Y':
-                if (!readValue(value, offset, lenStr, str, 4))
+                if (!readExactLengthValue(value, offset, lenStr, str, 4))
```

Comment:
How about the year 897?  Does it have to be written as 0897?

Reply:
Wow, what an obvious scenario I missed!  Thanks for finding this.

This (correctly) forces a re-think of the solution.  Most likely, %Y will have to go back to parsing 1-4 digits but it will be interpreted (stashed into struct tm) differently, while %y will be interpreted as a 2-digit shorthand for a year near the current year.

I will make that change and resubmit.  Probably the majority of the code just reviewed will change or be deleted.

Reply:
Not sure we support any year prior to 1900 anyway, do we?